release/ray_release/job_manager/anyscale_job_manager.py

```diff
@@ -55,6 +55,10 @@ def _run_job(
         upload_path: Optional[str] = None,
         pip: Optional[List[str]] = None,
     ) -> None:
+        env_vars[
+            "ANYSCALE_JOB_CLUSTER_ENV_NAME"
+        ] = self.cluster_manager.cluster_env_name
+
```

Comment:
env_vars is a dict that is sent in, I do not think you are suppose to change its content.

could you make a copy of `env_vars` and then set the `ANYSCALE_JOB_CLUSTER_ENV_NAME`?

not sure, but maybe also add a unit test? (but if you cannot find it, it is fine.. @can-anyscale or I can add it later..)


also, I think the `env` related two lines can just be removed:

````
env = os.environ.copy()
env.setdefault("ANYSCALE_HOST", str(ANYSCALE_HOST))
````

those two lines do not seem to do anything useful...

Reply:
Sounds good, I've changed it to modify + pass in a copy of the env vars. I tried to add a unit test to `test_anyscale_job_manager.py` but couldn't find a way to execute it (kept getting `ModuleNotFoundError: No module named 'ray_release'` error)